packages/appeals-service-api/src/routes/v2/users/_id/appeal-submissions/_appealSubmissionId/service.js

```diff
@@ -0,0 +1,36 @@
+const { UserAppealSubmissionRepository } = require('./repo');
+const { AppealsRepository } = require('#repositories/appeals-repository');
+
+const repo = new UserAppealSubmissionRepository();
+const cosmosAppeals = new AppealsRepository();
+
+/**
+ * @typedef {import("@prisma/client").AppealCase} AppealCase
+ * // TODO: define type for submission (ideally generated from spec)
+ * @typedef {AppealCase | any} AppealCaseOrSubmission
+ */
+
+/**
+ * Get all appeals for a user - both cases and draft submissions
+ *
+ * @param {{ userId: string, appealSubmissionId: string }} params
+ * @return {Promise<AppealCaseOrSubmission[]|null>}
+ */
+async function getAppealSubmissionForUser(params) {
+	const user = await repo.getAppealSubmissionForUser(params);
+	if (!user) {
+		return null;
+	}
+
+	const submissionBelongsToUser = user.Appeals.some(
+		({ Appeal }) => Appeal.legacyAppealSubmissionId === params.appealSubmissionId
+	);
```

Comment:
Not thrilled about this but I couldn't make prisma happy querying for legacySubmissionId

Reply:
Do we need user record, is AppealToUser enough?
So either AppealToUser + include Appeal
or:
Appeal + Include AppealToUser

so not entirely sure what prisma does with the below if the include comes back null but something like:

```js
return await this.dbClient.appeal.findFirst({
				where: {
					legacyAppealSubmissionId: legacyId
				},
				include: {
					Users: {
						where: {
							userId: userId,
							role: 'appellant' // seems sensible to check role too
						}
					}
				}
			});
```

Reply:
This worked great, thanks!